### PR TITLE
chore: Use the same cacheID and llb.Scratch()

### LINKED ISSUE #1357: feat(cache): Reuse python pip cache in all projects
## Description

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by gaocegege**
Fix #1357 

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/lang/ir/v0/cache.go [hunk 1]
```diff
@@ -15,18 +15,11 @@
...
 )
 
 func (g generalGraph) CacheID(filename string) string {
-	var cacheID string
-	if g.CUDA != nil {
-		cacheID = fmt.Sprintf("%s/%s-gpu", filename, g.EnvironmentName)
-	} else {
-		cacheID = fmt.Sprintf("%s/%s-cpu", filename, g.EnvironmentName)
-	}
-	logrus.Debugf("apt/pypi calculated cacheID: %s", cacheID)
```
Comment (kemingy) [inline pkg/lang/ir/v0/cache.go:30]: Does this also work for PyTorch? The CPU version and GPU version have exact the same name.

Comment (VoVAllen) [reply pkg/lang/ir/v0/cache.go:30]: Then have different names, such as "2.0" vs "2.0+cu113"

Comment (gaocegege) [reply pkg/lang/ir/v0/cache.go:30]: SGTM.

**Review by kemingy — APPROVED** [review]
/lgtm

**Review by VoVAllen — APPROVED** [review]
/lgtm

**Pull request merged**